huey/tests/queue.py

```diff
@@ -228,6 +235,21 @@ def test_execution(self):
 
         self.assertRaises(TypeError, huey.execute, huey.dequeue())
 
+    def test_self_awareness(self):
+        self_aware('k', 'v')
+        task = huey.dequeue()
+        huey.execute(task)
+        self.assertEqual(last_executed_task_class.pop(), "queuecmd_self_aware")
+
+        self_aware('k', 'v')
+        huey.execute(huey.dequeue())
+        self.assertEqual(last_executed_task_class.pop(), "queuecmd_self_aware")
```

Comment:
Any reason this is called twice? Seems to be doing the same thing as the previous block.